pkg/platform/runtime/download.go

```diff
@@ -123,33 +123,47 @@ func (r *Download) fetchRecipeID() (strfmt.UUID, *failures.Failure) {
 // FetchArtifacts will retrieve artifact information from the head-chef (eg language installers)
 // The first return argument specifies whether we are dealing with an alternative build
 func (r *Download) FetchArtifacts() (*FetchArtifactsResult, *failures.Failure) {
-	platProject, fail := model.FetchProjectByName(r.owner, r.projectName)
-	if fail != nil {
-		return nil, fail
-	}
-	r.orgID = platProject.OrganizationID.String()
-	r.private = platProject.Private
+	commitID := strfmt.UUID(constants.ValidZeroUUID)
+	recipeID := strfmt.UUID(constants.ValidZeroUUID)
+	orgID := strfmt.UUID(constants.ValidZeroUUID)
+	projectID := strfmt.UUID(constants.ValidZeroUUID)
+	var fail *failures.Failure
+
+	if r.commitID != "" {
+		commitID = r.commitID
+	} else {
+		platProject, fail := model.FetchProjectByName(r.owner, r.projectName)
+		if fail != nil {
+			return nil, fail
+		}
 
-	recipeID, fail := r.fetchRecipeID()
-	if fail != nil {
-		return nil, fail
-	}
+		projectID = platProject.ProjectID
+		orgID = platProject.OrganizationID
+		r.orgID = platProject.OrganizationID.String()
+		r.private = platProject.Private
 
-	var commitID *strfmt.UUID
-	for _, branch := range platProject.Branches {
-		if branch.Default {
-			commitID = branch.CommitID
-			break
+		var branchCommitID *strfmt.UUID
+		for _, branch := range platProject.Branches {
+			if branch.Default {
+				branchCommitID = branch.CommitID
+				break
+			}
+		}
+		if branchCommitID == nil {
+			return nil, FailNoCommitID.New("fetch_err_runtime_no_commitid")
 		}
+		commitID = *branchCommitID
```

Comment:
I just realized this code is bugged / out of date (as in, the code you're updating). There is no scenario in which we don't feed the commitID into the runtime downloader. So if the commitID was not given this would effectively amount to an error.

You can drop this whole chunk in favour of line 133.

Reply:
I dropped most of this function. Even setting the orgID and projectID may be useless, but I suspect it would be worth passing that into the build request if we can.